lib/Model/Acl.php

```diff
@@ -316,6 +323,18 @@ public function getAllowEdit(): bool {
 		return ($this->getIsOwner() || $this->getIsAdmin());
 	}
 
+	/**
+	 * @NoAdminRequired
+	 * @return bool
+	 */
+	public function getAllowAddEvent(): bool {
+		return (
+			   $this->getAllowEdit()
+			&& $this->poll->getType() === 'datePoll'
+			&& $this->appManager->isEnabledForUser('calendar')
+		);
+	}
+
```

Comment:
@tortuetorche I added a new check to the acl: getAllowAddEvent().
It checks the access to the calendar app and integrates your condition.